### PR TITLE
Add helm hook for custom secret validation

### PR DESCRIPTION
**Pull request by dmitryax**
Depends on https://github.com/signalfx/splunk-otel-collector-chart/pull/295

### CODE REVIEW

File: helm-charts/splunk-otel-collector/templates/secret-validation-hook.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  annotations:
+    "helm.sh/hook": pre-upgrade,pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  restartPolicy: Never
+  containers:
+  - name: validate-secret
+    image: {{ template "splunk-otel-collector.image.otelcol" . }}
+    imagePullPolicy: {{ .Values.image.otelcol.pullPolicy }}
+    command: ["sh", "-c"]
```
Comment (rockb1017) [inline helm-charts/splunk-otel-collector/templates/secret-validation-hook.yaml:19]: this image doesn't have `sh`, no?

Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/secret-validation-hook.yaml:19]: It does have it

Comment (rockb1017) [reply helm-charts/splunk-otel-collector/templates/secret-validation-hook.yaml:19]: oh i see it does. but how? just curious.

**Review by rockb1017 — APPROVED** [review]

File: helm-charts/splunk-otel-collector/templates/secret-validation-hook.yaml (hunk 1 above)
Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/secret-validation-hook.yaml:19]: It's being picked up from signalfx-agent bundle https://github.com/signalfx/signalfx-agent/blob/main/Dockerfile#L306-L332

**Pull request merged**